website_sale_contract/wizards/contract_checkout.py

```diff
@@ -0,0 +1,68 @@
+# -*- coding: utf-8 -*-
+# Copyright 2017 LasLabs Inc.
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl.html).
+
+from odoo import api, fields, models
+
+
+class ContractCheckout(models.TransientModel):
+
+    _name = 'contract.checkout'
+    _description = 'Contract Checkout'
+
+    order_id = fields.Many2one(
+        string='Order',
+        comodel_name='sale.order',
+        default=lambda s: s._default_order_id(),
+        required=True,
+    )
+    contract_ids = fields.One2many(
+        string='Contracts',
+        comodel_name='contract.temp',
+        inverse_name='contract_checkout_id',
+    )
+
+    @api.model
+    def _default_order_id(self):
+        if self.env.context.get('active_model') != 'sale.order':
+            return
+        return self.env['sale.order'].browse(
+            self.env.context.get('active_id')
+        )
+
+    @api.multi
+    def _create_temp_lines(self, sale_order_lines):
+        self.ensure_one()
+        ContractTemp = self.env['contract.temp']
```

Comment:
This seems pretty inefficient & I'm not fully understanding what you're going for. Is this so that you can create a linked list type hierarchy of the contracts for iteration? Is there any other point to the temp contracts?